src/pyownet/protocol.py

```diff
@@ -509,19 +532,20 @@ def dir(self, path='/', slash=True, bus=False):
         else:
             return []
 
-    def read(self, path, size=MAX_PAYLOAD, offset=0):
```

Comment:
I think that it is better not to break the current API: `timeout=0` should always be the last argument. (Major changes should be delayed to forthcoming version 1.0.0)